src/wallet/rpcwallet.cpp

```diff
@@ -37,6 +37,7 @@ static const std::string WALLET_ENDPOINT_BASE = "/wallet/";
 
 CWallet *GetWalletForJSONRPCRequest(const JSONRPCRequest& request)
 {
+    AssertLockHeld(cs_wallets);
```

Comment:
Why not just `LOCK(cs_wallets)` too?

Reply:
Because I'd prefer to avoid recursive locking if possible.

Reply:
Just curious, why?

Reply:
Yes, definitely. cs_wallets is locked *before* cs_main in global lock order, so putting it too many places is just asking for trouble, IMO.